### PR TITLE
dird: make UA tree command auditing possible

### PR DESCRIPTION
**Pull request by fbergkemper**
1. Introduce a new boolean named audit_event in the struct cmdstruct
of the TreeContext to mark commands which require auditing.

2. Introduce a condition in the UserSelectFilesFromTree function to
call LogAuditEventCmdline if the given command wants an audit event.

### CODE REVIEW

File: core/src/dird/ua_restore.cc [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
    Copyright (C) 2002-2011 Free Software Foundation Europe e.V.
    Copyright (C) 2011-2016 Planets Communications B.V.
-   Copyright (C) 2013-2019 Bareos GmbH & Co. KG
+   Copyright (C) 2013-2020 Bareos GmbH & Co. KG
```
Comment (franku) [inline core/src/dird/ua_restore.cc:6]: This should be placed in the next commit..

File: core/src/dird/ua_audit.cc [hunk 2]
```diff
@@ -142,4 +142,20 @@ void UaContext::LogAuditEventCmdline()
   Emsg3(M_AUDIT, 0, _("Console [%s] from [%s] cmdline %s\n"), user_name, host,
         cmd);
 }
+
+void UaContext::LogAuditEventInfoMsg(const char* audit_msg, const char* item)
```
Comment (franku) [inline core/src/dird/ua_audit.cc:146]:
```suggestion
void UaContext::LogAuditEventInfoMsg(const char* fmt, ...)
```

Comment (franku) [reply core/src/dird/ua_audit.cc:146]: Would be nice to have an interface that has a format string as input parameter. 

See void UaContext::InfoMsg(const char* fmt, ...) as example.

File: core/src/dird/ua_run.cc [hunk 3]
```diff
@@ -525,6 +525,15 @@ int DoRunCmd(UaContext* ua, const char* cmd)
 
     FreeJcr(jcr); /* release jcr */
 
+    /*
+     * For interactive runs we send a message to the audit log
+     */
+    if (jcr->impl->IgnoreLevelPoolOverides) {
```
Comment (franku) [inline core/src/dird/ua_run.cc:531]:
```suggestion
    if (jcr->impl->IgnoreLevelPoolOverrides) {
```

Comment (franku) [reply core/src/dird/ua_run.cc:531]: Although this typo is not yours, you may want to correct that.

File: core/src/dird/ua_audit.cc (hunk 2 above)
Comment (fbergkemper) [reply core/src/dird/ua_audit.cc]: Suggestion implemented

File: core/src/dird/ua_run.cc (hunk 3 above)
Comment (fbergkemper) [reply core/src/dird/ua_run.cc]: corrected spelling error

File: core/src/dird/ua_restore.cc (hunk 1 above)
Comment (fbergkemper) [reply core/src/dird/ua_restore.cc:6]: rebase accident corrected

**Pull request merged**